src/Runners/PHPUnit/SuiteLoader.php

```diff
@@ -263,7 +264,17 @@ private function getMethodTests(ParsedClass $class, ParsedFunction $method): arr
             $testFullClassName = '\\' . $class->getName();
             $testClass         = new $testFullClassName();
             $result            = [];
-            foreach ($testClass->$dataProvider() as $key => $value) {
+
+            $testClassReflection = new ReflectionClass($testFullClassName);
+            $dataProviderMethod  = $testClassReflection->getMethod($dataProvider);
+
+            if ($dataProviderMethod->getNumberOfParameters() === 0) {
+                $data = $dataProviderMethod->invoke($testClass);
+            } else {
+                $data = $dataProviderMethod->invoke($testClass, $method->getName());
```

Comment:
I guess you are referring to this: https://github.com/sebastianbergmann/phpunit/blob/9.2.6/src/Util/Annotation/DocBlock.php#L389-L393

Could you please add a test for this?